# This file is part of HiGFXback

# requires
REQUIRES="directfb2-media-build luafilesystem-build make-build"

PKG_CONFIG_PATH=/dfb/share/pkgconfig pkg-config --exists --print-errors $REQUIRES || exit 1

# configure
sh ./Configure --with-dfb-include /dfb/include/directfb --with-dfb-lib /dfb/lib --prefix /dfb

# build
make DISPLAY_DRIVER=directfb all

# install
install -d $DESTDIR/dfb/lib/lua/5.1/tek/lib
install tek/lib/exec.so $DESTDIR/dfb/lib/lua/5.1/tek/lib
install tek/lib/region.so $DESTDIR/dfb/lib/lua/5.1/tek/lib
install tek/lib/string.so $DESTDIR/dfb/lib/lua/5.1/tek/lib
install tek/lib/support.so $DESTDIR/dfb/lib/lua/5.1/tek/lib
install tek/lib/visual.so $DESTDIR/dfb/lib/lua/5.1/tek/lib
install -d $DESTDIR/dfb/lib/lua/5.1/tek/lib/display
install tek/lib/display/directfb.so $DESTDIR/dfb/lib/lua/5.1/tek/lib/display
install -d $DESTDIR/dfb/lib/lua/5.1/tek/ui/class
install tek/ui/class/area.so $DESTDIR/dfb/lib/lua/5.1/tek/ui/class
install tek/ui/class/frame.so $DESTDIR/dfb/lib/lua/5.1/tek/ui/class
install -d $DESTDIR/dfb/lib/lua/5.1/tek/ui/layout
install tek/ui/layout/default.so $DESTDIR/dfb/lib/lua/5.1/tek/ui/layout
install -d $DESTDIR/dfb/share/lua/5.1
install bin/childtask.lua $DESTDIR/dfb/share/lua/5.1/tekui_childtask.lua
install bin/demo.lua $DESTDIR/dfb/share/lua/5.1/tekui_demo.lua
install bin/demo_alignment.lua $DESTDIR/dfb/share/lua/5.1/tekui_demo_alignment.lua
install bin/demo_animations.lua $DESTDIR/dfb/share/lua/5.1/tekui_demo_animations.lua
install bin/demo_borders.lua $DESTDIR/dfb/share/lua/5.1/tekui_demo_borders.lua
install bin/demo_choices.lua $DESTDIR/dfb/share/lua/5.1/tekui_demo_choices.lua
install bin/demo_floattext.lua $DESTDIR/dfb/share/lua/5.1/tekui_demo_floattext.lua
install bin/demo_gradient.lua $DESTDIR/dfb/share/lua/5.1/tekui_demo_gradient.lua
install bin/demo_graphics.lua $DESTDIR/dfb/share/lua/5.1/tekui_demo_graphics.lua
install bin/demo_input.lua $DESTDIR/dfb/share/lua/5.1/tekui_demo_input.lua
install bin/demo_layout.lua $DESTDIR/dfb/share/lua/5.1/tekui_demo_layout.lua
install bin/demo_list.lua $DESTDIR/dfb/share/lua/5.1/tekui_demo_list.lua
install bin/demo_popups.lua $DESTDIR/dfb/share/lua/5.1/tekui_demo_popups.lua
install bin/demo_presentation.lua $DESTDIR/dfb/share/lua/5.1/tekui_demo_presentation.lua
install bin/demo_slider.lua $DESTDIR/dfb/share/lua/5.1/tekui_demo_slider.lua
install bin/editor.lua $DESTDIR/dfb/share/lua/5.1/tekui_editor.lua
install bin/meter.lua $DESTDIR/dfb/share/lua/5.1/tekui_meter.lua
install bin/textlist.lua $DESTDIR/dfb/share/lua/5.1/tekui_textlist.lua
install bin/tsp.lua $DESTDIR/dfb/share/lua/5.1/tekui_tsp.lua
install -d $DESTDIR/dfb/share/lua/5.1/graphics
install -m 644 bin/graphics/locale.ppm $DESTDIR/dfb/share/lua/5.1/graphics
install -m 644 bin/graphics/locale_alpha.png $DESTDIR/dfb/share/lua/5.1/graphics
install -m 644 bin/graphics/world.ppm $DESTDIR/dfb/share/lua/5.1/graphics
install -d $DESTDIR/dfb/share/lua/5.1/tek
install -m 644 tek/class.lua $DESTDIR/dfb/share/lua/5.1/tek
install -m 644 tek/ui.lua $DESTDIR/dfb/share/lua/5.1/tek
install -d $DESTDIR/dfb/share/lua/5.1/tek/class
install -m 644 tek/class/list.lua $DESTDIR/dfb/share/lua/5.1/tek/class
install -m 644 tek/class/object.lua $DESTDIR/dfb/share/lua/5.1/tek/class
install -d $DESTDIR/dfb/share/lua/5.1/tek/lib
install -m 644 tek/lib/args.lua $DESTDIR/dfb/share/lua/5.1/tek/lib
install -m 644 tek/lib/debug.lua $DESTDIR/dfb/share/lua/5.1/tek/lib
install -d $DESTDIR/dfb/share/lua/5.1/tek/ui/border
install -m 644 tek/ui/border/default.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/border
install -d $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 bin/tek/ui/class/boing.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 bin/tek/ui/class/editwindow.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 bin/tek/ui/class/meter.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 bin/tek/ui/class/plasma.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 bin/tek/ui/class/textlist.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 bin/tek/ui/class/tunnel.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 tek/ui/class/application.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 tek/ui/class/border.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 tek/ui/class/button.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 tek/ui/class/canvas.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 tek/ui/class/checkmark.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 tek/ui/class/dirlist.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 tek/ui/class/display.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 tek/ui/class/drawhook.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 tek/ui/class/element.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 tek/ui/class/family.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 tek/ui/class/floattext.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 tek/ui/class/gauge.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 tek/ui/class/group.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 tek/ui/class/handle.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 tek/ui/class/image.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 tek/ui/class/imagewidget.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 tek/ui/class/input.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 tek/ui/class/layout.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 tek/ui/class/lister.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 tek/ui/class/listview.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 tek/ui/class/menuitem.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 tek/ui/class/numeric.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 tek/ui/class/pagegroup.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 tek/ui/class/popitem.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 tek/ui/class/poplist.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 tek/ui/class/popupwindow.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 tek/ui/class/radiobutton.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 tek/ui/class/scrollbar.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 tek/ui/class/scrollgroup.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 tek/ui/class/sizeable.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 tek/ui/class/slider.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 tek/ui/class/spacer.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 tek/ui/class/text.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 tek/ui/class/textedit.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 tek/ui/class/widget.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -m 644 tek/ui/class/window.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/class
install -d $DESTDIR/dfb/share/lua/5.1/tek/ui/cursor
install -m 644 tek/ui/cursor/cursor-green.png /dfb/share/lua/5.1/tek/ui/cursor
install -d $DESTDIR/dfb/share/lua/5.1/tek/ui/font
install -m 644 tek/ui/font/Vera.ttf $DESTDIR/dfb/share/lua/5.1/tek/ui/font
install -m 644 tek/ui/font/VeraMono.ttf $DESTDIR/dfb/share/lua/5.1/tek/ui/font
install -d $DESTDIR/dfb/share/lua/5.1/tek/ui/hook
install -m 644 tek/ui/hook/ripple.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/hook
install -d $DESTDIR/dfb/share/lua/5.1/tek/ui/image
install -m 644 tek/ui/image/arrowdown.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/image
install -m 644 tek/ui/image/arrowleft.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/image
install -m 644 tek/ui/image/arrowright.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/image
install -m 644 tek/ui/image/arrowup.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/image
install -m 644 tek/ui/image/checkmark.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/image
install -m 644 tek/ui/image/file.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/image
install -m 644 tek/ui/image/radiobutton.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/image
install -d $DESTDIR/dfb/share/lua/5.1/tek/ui/locale/schulze-mueller.de/dirlist-class
install -m 644 tek/ui/locale/schulze-mueller.de/dirlist-class/en $DESTDIR/dfb/share/lua/5.1/tek/ui/locale/schulze-mueller.de/dirlist-class
install -d $DESTDIR/dfb/share/lua/5.1/tek/ui/locale/schulze-mueller.de/editwindow-class
install -m 644 bin/tek/ui/locale/schulze-mueller.de/editwindow-class/en $DESTDIR/dfb/share/lua/5.1/tek/ui/locale/schulze-mueller.de/editwindow-class
install -d $DESTDIR/dfb/share/lua/5.1/tek/ui/locale/schulze-mueller.de/tekui-demo
install -m 644 bin/tek/ui/locale/schulze-mueller.de/tekui-demo/en $DESTDIR/dfb/share/lua/5.1/tek/ui/locale/schulze-mueller.de/tekui-demo
install -d $DESTDIR/dfb/share/lua/5.1/tek/ui/style
install -m 644 bin/tek/ui/style/editor.css $DESTDIR/dfb/share/lua/5.1/tek/ui/style
install -m 644 tek/ui/style/default.css $DESTDIR/dfb/share/lua/5.1/tek/ui/style
install -m 644 tek/ui/style/desktop.css $DESTDIR/dfb/share/lua/5.1/tek/ui/style
install -m 644 tek/ui/style/desktop.lua $DESTDIR/dfb/share/lua/5.1/tek/ui/style
install -m 644 tek/ui/style/user.css $DESTDIR/dfb/share/lua/5.1/tek/ui/style

# build.pc
install -d $DESTDIR/dfb/share/pkgconfig
cat > $DESTDIR/dfb/share/pkgconfig/tekui-build.pc << EOF
Name: tekui
Version: 1.11-r1
Description: tekUI Toolkit
Requires: $REQUIRES

exec=\\
/dfb/lib/lua/5.1/tek/lib/exec.so \\
/dfb/lib/lua/5.1/tek/lib/region.so \\
/dfb/lib/lua/5.1/tek/lib/string.so \\
/dfb/lib/lua/5.1/tek/lib/support.so \\
/dfb/lib/lua/5.1/tek/lib/visual.so \\
/dfb/lib/lua/5.1/tek/lib/display/directfb.so \\
/dfb/lib/lua/5.1/tek/ui/class/area.so \\
/dfb/lib/lua/5.1/tek/ui/class/frame.so \\
/dfb/lib/lua/5.1/tek/ui/layout/default.so \\
/dfb/share/lua/5.1/tekui_childtask.lua \\
/dfb/share/lua/5.1/tekui_demo.lua \\
/dfb/share/lua/5.1/tekui_demo_alignment.lua \\
/dfb/share/lua/5.1/tekui_demo_animations.lua \\
/dfb/share/lua/5.1/tekui_demo_borders.lua \\
/dfb/share/lua/5.1/tekui_demo_choices.lua \\
/dfb/share/lua/5.1/tekui_demo_floattext.lua \\
/dfb/share/lua/5.1/tekui_demo_gradient.lua \\
/dfb/share/lua/5.1/tekui_demo_graphics.lua \\
/dfb/share/lua/5.1/tekui_demo_input.lua \\
/dfb/share/lua/5.1/tekui_demo_layout.lua \\
/dfb/share/lua/5.1/tekui_demo_list.lua \\
/dfb/share/lua/5.1/tekui_demo_popups.lua \\
/dfb/share/lua/5.1/tekui_demo_presentation.lua \\
/dfb/share/lua/5.1/tekui_demo_slider.lua \\
/dfb/share/lua/5.1/tekui_editor.lua \\
/dfb/share/lua/5.1/tekui_meter.lua \\
/dfb/share/lua/5.1/tekui_textlist.lua \\
/dfb/share/lua/5.1/tekui_tsp.lua \\
/dfb/share/lua/5.1/graphics/locale.ppm \\
/dfb/share/lua/5.1/graphics/locale_alpha.png \\
/dfb/share/lua/5.1/graphics/world.ppm \\
/dfb/share/lua/5.1/tek/class.lua \\
/dfb/share/lua/5.1/tek/class/list.lua \\
/dfb/share/lua/5.1/tek/class/object.lua \\
/dfb/share/lua/5.1/tek/lib/args.lua \\
/dfb/share/lua/5.1/tek/lib/debug.lua \\
/dfb/share/lua/5.1/tek/ui.lua \\
/dfb/share/lua/5.1/tek/ui/border/default.lua \\
/dfb/share/lua/5.1/tek/ui/class/application.lua \\
/dfb/share/lua/5.1/tek/ui/class/boing.lua \\
/dfb/share/lua/5.1/tek/ui/class/border.lua \\
/dfb/share/lua/5.1/tek/ui/class/button.lua \\
/dfb/share/lua/5.1/tek/ui/class/canvas.lua \\
/dfb/share/lua/5.1/tek/ui/class/checkmark.lua \\
/dfb/share/lua/5.1/tek/ui/class/dirlist.lua \\
/dfb/share/lua/5.1/tek/ui/class/display.lua \\
/dfb/share/lua/5.1/tek/ui/class/drawhook.lua \\
/dfb/share/lua/5.1/tek/ui/class/editwindow.lua \\
/dfb/share/lua/5.1/tek/ui/class/element.lua \\
/dfb/share/lua/5.1/tek/ui/class/family.lua \\
/dfb/share/lua/5.1/tek/ui/class/floattext.lua \\
/dfb/share/lua/5.1/tek/ui/class/gauge.lua \\
/dfb/share/lua/5.1/tek/ui/class/handle.lua \\
/dfb/share/lua/5.1/tek/ui/class/image.lua \\
/dfb/share/lua/5.1/tek/ui/class/imagewidget.lua \\
/dfb/share/lua/5.1/tek/ui/class/input.lua \\
/dfb/share/lua/5.1/tek/ui/class/layout.lua \\
/dfb/share/lua/5.1/tek/ui/class/lister.lua \\
/dfb/share/lua/5.1/tek/ui/class/listview.lua \\
/dfb/share/lua/5.1/tek/ui/class/menuitem.lua \\
/dfb/share/lua/5.1/tek/ui/class/meter.lua \\
/dfb/share/lua/5.1/tek/ui/class/numeric.lua \\
/dfb/share/lua/5.1/tek/ui/class/pagegroup.lua \\
/dfb/share/lua/5.1/tek/ui/class/plasma.lua \\
/dfb/share/lua/5.1/tek/ui/class/popitem.lua \\
/dfb/share/lua/5.1/tek/ui/class/poplist.lua \\
/dfb/share/lua/5.1/tek/ui/class/popupwindow.lua \\
/dfb/share/lua/5.1/tek/ui/class/radiobutton.lua \\
/dfb/share/lua/5.1/tek/ui/class/scrollbar.lua \\
/dfb/share/lua/5.1/tek/ui/class/scrollgroup.lua \\
/dfb/share/lua/5.1/tek/ui/class/sizeable.lua \\
/dfb/share/lua/5.1/tek/ui/class/slider.lua \\
/dfb/share/lua/5.1/tek/ui/class/spacer.lua \\
/dfb/share/lua/5.1/tek/ui/class/text.lua \\
/dfb/share/lua/5.1/tek/ui/class/textlist.lua \\
/dfb/share/lua/5.1/tek/ui/class/textedit.lua \\
/dfb/share/lua/5.1/tek/ui/class/tunnel.lua \\
/dfb/share/lua/5.1/tek/ui/class/widget.lua \\
/dfb/share/lua/5.1/tek/ui/class/window.lua \\
/dfb/share/lua/5.1/tek/ui/cursor/cursor-green.png \\
/dfb/share/lua/5.1/tek/ui/font/Vera.ttf \\
/dfb/share/lua/5.1/tek/ui/font/VeraMono.ttf \\
/dfb/share/lua/5.1/tek/ui/hook/ripple.lua \\
/dfb/share/lua/5.1/tek/ui/image/arrowdown.lua \\
/dfb/share/lua/5.1/tek/ui/image/arrowleft.lua \\
/dfb/share/lua/5.1/tek/ui/image/arrowright.lua \\
/dfb/share/lua/5.1/tek/ui/image/arrowup.lua \\
/dfb/share/lua/5.1/tek/ui/image/checkmark.lua \\
/dfb/share/lua/5.1/tek/ui/image/file.lua \\
/dfb/share/lua/5.1/tek/ui/image/radiobutton.lua \\
/dfb/share/lua/5.1/tek/ui/locale/schulze-mueller.de/dirlist-class/en \\
/dfb/share/lua/5.1/tek/ui/locale/schulze-mueller.de/editwindow-class/en \\
/dfb/share/lua/5.1/tek/ui/locale/schulze-mueller.de/tekui-demo/en \\
/dfb/share/lua/5.1/tek/ui/style/default.css \\
/dfb/share/lua/5.1/tek/ui/style/desktop.css \\
/dfb/share/lua/5.1/tek/ui/style/desktop.lua \\
/dfb/share/lua/5.1/tek/ui/style/editor.css \\
/dfb/share/lua/5.1/tek/ui/style/user.css
EOF

exit
--- tekui-1.11-r1.orig/Configure
+++ tekui-1.11-r1/Configure
@@ -0,0 +1,36 @@
+#!/bin/sh
+
+PREFIX=/usr/local
+
+DIRECTFB_DEFS="\`pkg-config --cflags directfb\`"
+DIRECTFB_LIBS="\`pkg-config --libs directfb\`"
+
+while : ; do
+  case "$1" in
+    "")
+      break
+    ;;
+    --prefix)
+      PREFIX=$2
+      shift
+    ;;
+    --with-dfb-include)
+      DIRECTFB_DEFS=-I$2
+      shift
+    ;;
+    --with-dfb-lib)
+      DIRECTFB_LIBS="-L$2 -Wl,-rpath,$2 -ldirectfb"
+      shift
+    ;;
+    *)
+      echo Unknown option \"$1\".
+      exit 1
+    ;;
+  esac
+  shift
+done
+
+sed -i "/^PREFIX ?=/s#=.*#= $PREFIX#" config
+
+sed -i "/^DIRECTFB_DEFS =/s#=.*#= $DIRECTFB_DEFS#" config
+sed -i "/^DIRECTFB_LIBS =/s#=.*#= $DIRECTFB_LIBS#" config
--- tekui-1.11-r1.orig/bin/bashing.lua
+++ tekui-1.11-r1/bin/bashing.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 local ui = require "tek.ui"
 
--- tekui-1.11-r1.orig/bin/buttonclass.lua
+++ tekui-1.11-r1/bin/buttonclass.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 ui = require "tek.ui"
 
--- tekui-1.11-r1.orig/bin/canvas.lua
+++ tekui-1.11-r1/bin/canvas.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 local ui = require "tek.ui"
 
--- tekui-1.11-r1.orig/bin/childprocess.lua
+++ tekui-1.11-r1/bin/childprocess.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 local posix = require "posix"
 local char = string.char
--- tekui-1.11-r1.orig/bin/childtask.lua
+++ tekui-1.11-r1/bin/childtask.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 local exec = require "tek.lib.exec"
 local guitask = exec.run(function()
 	-- child task:
@@ -32,7 +32,7 @@
 If you click on the button below, the command will be sent
 to the application's parent task. 
 
-Clicking the close button will send the command 'quit'.]]
+The 'quit' command will close the application.]]
 							},
 							ui.Group:new 
 							{
--- tekui-1.11-r1.orig/bin/compiler.lua
+++ tekui-1.11-r1/bin/compiler.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 --
 --	compiler.lua - Lua compiler and module linker
--- tekui-1.11-r1.orig/bin/demo.lua
+++ tekui-1.11-r1/bin/demo.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 --
 --	bin/demo.lua - tekUI demo
 --	Written by Timm S. Mueller <tmueller at schulze-mueller.de>
@@ -57,7 +57,7 @@
 	local demos = { }
 
 	for fname in lfs.readdir(ui.ProgDir) do
-		if fname:match("^demo_.*%.lua$") then
+		if fname:match("^demo_.*%.lua$") or fname:match("^tekui_demo_.*%.lua$") then
 			fname = ui.ProgDir .. fname
 			db.info("Loading demo '%s' ...", fname)
 			local success, res = pcall(dofile, fname)
--- tekui-1.11-r1.orig/bin/demo_alignment.lua
+++ tekui-1.11-r1/bin/demo_alignment.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 local ui = require "tek.ui"
 local List = require "tek.class.list"
--- tekui-1.11-r1.orig/bin/demo_animations.lua
+++ tekui-1.11-r1/bin/demo_animations.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 local ui = require "tek.ui"
 local Group = ui.Group
--- tekui-1.11-r1.orig/bin/demo_borders.lua
+++ tekui-1.11-r1/bin/demo_borders.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 local ui = require "tek.ui"
 local Group = ui.Group
--- tekui-1.11-r1.orig/bin/demo_buttons.lua
+++ tekui-1.11-r1/bin/demo_buttons.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 local ui = require "tek.ui"
 local Button = ui.Button
--- tekui-1.11-r1.orig/bin/demo_choices.lua
+++ tekui-1.11-r1/bin/demo_choices.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 local ui = require "tek.ui"
 local Group = ui.Group
--- tekui-1.11-r1.orig/bin/demo_floattext.lua
+++ tekui-1.11-r1/bin/demo_floattext.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 local ui = require "tek.ui"
 local L = ui.getLocale("tekui-demo", "schulze-mueller.de")
--- tekui-1.11-r1.orig/bin/demo_gradient.lua
+++ tekui-1.11-r1/bin/demo_gradient.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 local ui = require "tek.ui"
 local Group = ui.Group
--- tekui-1.11-r1.orig/bin/demo_graphics.lua
+++ tekui-1.11-r1/bin/demo_graphics.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 local ui = require "tek.ui"
 
--- tekui-1.11-r1.orig/bin/demo_input.lua
+++ tekui-1.11-r1/bin/demo_input.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 local ui = require "tek.ui"
 local Button = ui.Button
--- tekui-1.11-r1.orig/bin/demo_layout.lua
+++ tekui-1.11-r1/bin/demo_layout.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 local ui = require "tek.ui"
 local Button = ui.Button
--- tekui-1.11-r1.orig/bin/demo_list.lua
+++ tekui-1.11-r1/bin/demo_list.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 local ui = require "tek.ui"
 local List = require "tek.class.list"
--- tekui-1.11-r1.orig/bin/demo_popups.lua
+++ tekui-1.11-r1/bin/demo_popups.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 local ui = require "tek.ui"
 local List = require "tek.class.list"
--- tekui-1.11-r1.orig/bin/demo_presentation.lua
+++ tekui-1.11-r1/bin/demo_presentation.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 local ui = require "tek.ui"
 local Group = ui.Group
--- tekui-1.11-r1.orig/bin/demo_slider.lua
+++ tekui-1.11-r1/bin/demo_slider.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 local ui = require "tek.ui"
 local Gauge = ui.Gauge
--- tekui-1.11-r1.orig/bin/dynamic.lua
+++ tekui-1.11-r1/bin/dynamic.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 local ui = require "tek.ui"
 ui.Application:new
 {
--- tekui-1.11-r1.orig/bin/editor.lua
+++ tekui-1.11-r1/bin/editor.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 local db = require "tek.lib.debug"
 local ui = require "tek.ui"
--- tekui-1.11-r1.orig/bin/filereq.lua
+++ tekui-1.11-r1/bin/filereq.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 local ui = require "tek.ui"
 
--- tekui-1.11-r1.orig/bin/fixed.lua
+++ tekui-1.11-r1/bin/fixed.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 local ui = require "tek.ui"
 
--- tekui-1.11-r1.orig/bin/gendata.lua
+++ tekui-1.11-r1/bin/gendata.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 local exec = require "tek.lib.exec"
 
--- tekui-1.11-r1.orig/bin/gendoc.lua
+++ tekui-1.11-r1/bin/gendoc.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 --
 --	tek/app/gendoc.lua - Document generator
--- tekui-1.11-r1.orig/bin/gridalign.lua
+++ tekui-1.11-r1/bin/gridalign.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 local ui = require "tek.ui"
 
--- tekui-1.11-r1.orig/bin/helloworld.lua
+++ tekui-1.11-r1/bin/helloworld.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 ui = require "tek.ui"
 
--- tekui-1.11-r1.orig/bin/imagetest.lua
+++ tekui-1.11-r1/bin/imagetest.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 local ui = require "tek.ui"
 
--- tekui-1.11-r1.orig/bin/layouthook.lua
+++ tekui-1.11-r1/bin/layouthook.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 local ui = require "tek.ui"
 local Region = require "tek.lib.region"
--- tekui-1.11-r1.orig/bin/meter.lua
+++ tekui-1.11-r1/bin/meter.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 local floor = math.floor
 ui = require "tek.ui"
--- tekui-1.11-r1.orig/bin/plasma.lua
+++ tekui-1.11-r1/bin/plasma.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 local visual = require "tek.lib.visual"
 
--- tekui-1.11-r1.orig/bin/runxml.lua
+++ tekui-1.11-r1/bin/runxml.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 --
 --	runxml.lua - XML to Lua+tekUI converter, runner, and checker
--- tekui-1.11-r1.orig/bin/shell.lua
+++ tekui-1.11-r1/bin/shell.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 local String = require "tek.lib.string"
 local ui = require "tek.ui"
--- tekui-1.11-r1.orig/bin/socketmessage.lua
+++ tekui-1.11-r1/bin/socketmessage.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 local socket
 pcall(function() socket = require "socket" end)
--- tekui-1.11-r1.orig/bin/style.lua
+++ tekui-1.11-r1/bin/style.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua 
+#!/bin/env lua 
 
 local db = require "tek.lib.debug"
 local ui = require "tek.ui"
--- tekui-1.11-r1.orig/bin/textlist.lua
+++ tekui-1.11-r1/bin/textlist.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 local String = require "tek.lib.string"
 local ui = require "tek.ui"
--- tekui-1.11-r1.orig/bin/tsp.lua
+++ tekui-1.11-r1/bin/tsp.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 --
 --	tsp.lua - Travelling salesman solver
--- tekui-1.11-r1.orig/bin/tutorial-1.lua
+++ tekui-1.11-r1/bin/tutorial-1.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 ui = require "tek.ui"
 
--- tekui-1.11-r1.orig/bin/tutorial-2.lua
+++ tekui-1.11-r1/bin/tutorial-2.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 ui = require "tek.ui"
 
--- tekui-1.11-r1.orig/bin/tutorial-3.lua
+++ tekui-1.11-r1/bin/tutorial-3.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 ui = require "tek.ui"
 
--- tekui-1.11-r1.orig/bin/tutorial-4.lua
+++ tekui-1.11-r1/bin/tutorial-4.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 ui = require "tek.ui"
 
--- tekui-1.11-r1.orig/bin/tutorial-5.lua
+++ tekui-1.11-r1/bin/tutorial-5.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 ui = require "tek.ui"
 
--- tekui-1.11-r1.orig/bin/updatetest.lua
+++ tekui-1.11-r1/bin/updatetest.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 local ui = require "tek.ui"
 ui.ThemeName = "dark tutorial"
--- tekui-1.11-r1.orig/bin/vgroup.lua
+++ tekui-1.11-r1/bin/vgroup.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 local ui = require "tek.ui"
 
--- tekui-1.11-r1.orig/config
+++ tekui-1.11-r1/config
@@ -202,7 +202,7 @@
 # - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 
 DIRECTFB_LIBS = `pkg-config --libs directfb`
-DIRECTFB_DEFS = $(FREETYPE_DEFS) `pkg-config --cflags directfb`
+DIRECTFB_DEFS = `pkg-config --cflags directfb`
 
 # - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 # Windows defines and libraries:
--- tekui-1.11-r1.orig/etc/keymap2c.lua
+++ tekui-1.11-r1/etc/keymap2c.lua
@@ -1,4 +1,4 @@
-#!/usr/bin/env lua
+#!/bin/env lua
 
 -- 
 --	linux keymap converter
--- tekui-1.11-r1.orig/include/tek/config/posix.h
+++ tekui-1.11-r1/include/tek/config/posix.h
@@ -64,7 +64,7 @@
 */
 
 #define TDEBUG_PLATFORM_PUTS(s) do { fputs(s, stderr); fflush(stderr); } while(0)
-#define TDEBUG_PLATFORM_FATAL() (abort(), 0)
+#define TDEBUG_PLATFORM_FATAL() abort()
 
 /*****************************************************************************/
 /*
--- tekui-1.11-r1.orig/tek/lib/display/directfb_lua.c
+++ tekui-1.11-r1/tek/lib/display/directfb_lua.c
@@ -74,7 +74,11 @@
 	exec = *(TAPTR *) lua_touserdata(L, -1);
 
 	/* register functions: */
+#if LUA_VERSION_NUM < 502
 	luaL_register(L, "tek.lib.display.directfb", libfuncs);
+#else
+	luaL_newlib(L, libfuncs);
+#endif
 	/* s: exectab, execbase, libtab */
 
 	/* create userdata: */
